internal/webhooks/clusterclass_test.go

```diff
@@ -1042,11 +1205,147 @@ func TestClusterClassValidation(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
-			webhook := &ClusterClass{}
+			// Sets up the fakeClient for the test case.
+			fakeClient := fake.NewClientBuilder().
+				WithObjects(tt.clusters...).
+				WithObjects(tt.machineDeployments...).
+				WithScheme(fakeScheme).
+				Build()
+
+			// Create the webhook and add the fakeClient as its client.
+			webhook := &ClusterClass{Client: fakeClient}
+
 			if tt.expectErr {
-				g.Expect(webhook.validate(tt.old, tt.in)).NotTo(Succeed())
+				g.Expect(webhook.validateRemovedMachineDeploymentClassesAreNotUsed(tt.oldClusterClass, tt.newClusterClass).ToAggregate()).NotTo(Succeed())
 			} else {
-				g.Expect(webhook.validate(tt.old, tt.in)).To(Succeed())
+				g.Expect(webhook.validateRemovedMachineDeploymentClassesAreNotUsed(tt.oldClusterClass, tt.newClusterClass).ToAggregate()).To(Succeed())
+			}
+		})
+	}
+}
+
+func TestClusterClass_ValidateDelete(t *testing.T) {
+	// NOTE: ClusterTopology feature flag is disabled by default, thus preventing to create or update ClusterClasses.
```

Comment:
Not sure if the comment makes sense in this context. Can you please also check the other occurences?

Reply:
Other occurences are fine as they either occur where the flag is set or in the specific feature flag test.